### DISCUSSION TITLE
[Feature] Download Transcoded Video

### DISCUSSION THREAD

**Discussion by epiller** [immich-app/immich#10451] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

It would be really useful to be able to download the already existing transcoded video directly from the interface.

### Platform

- [ ] Server
- [X] Web
- [X] Mobile

**Comment by dionorgua** [comment]
I agree here.. In addition to this it would be cool to have a way to download or share transcoded video from mobile app due to followed reasons:
- Transcoded videos are usually much smaller (so faster downloads, less traffic)
- Typical usage scenarios for "share" is to share via social apps/messengers, where transcoded quality is more than enough
- Sometimes original video uses badly supported codec/container format (too old like MJPG or too new  like AV1)

**Reply by epiller** [reply]
Exactly. Especially since it's a wasteful process, where Immich will currently give me the full resolution video, then the app will convert it to crap quality in order to save bandwidth.

**Comment by dionorgua** [comment]
Related: https://github.com/immich-app/immich/discussions/15033

**Comment by ojsef39** [comment]
This is a critical feature for me, as the normal person will not get that weird .VOB files won't work via WhatsApp, but they will still try to share those, so it would be nice if the download button would download a version that is actually compatible with the client that downloads the file or give an option in which format it should download it.

**Comment by ensingerphilipp** [comment]
This is still a real gap im stumbling upon right now:

Wanting to share a video via whatsapp -> quality is bad text unreadable because of the weird original super highres video.

A well transcoded version would share easily, but now i have to setup ffmpeg fresh  on a pc again where immich already does the transcoding when needed but has no download option

**Comment by CarlvO** [comment]
Even though it cannot be done from the Immich controls, it _is_ possible to download the transcoded video, at least in Firefox.

Play the transcoded video, right click in the video window and select ´Save video as´. It will save as 'playback.mp4' (at least with my settings).
